gmloader/main.cpp

```diff
@@ -103,25 +109,45 @@ static fs::path get_absolute_path(const char* path, fs::path work_dir){
 
 int main(int argc, char *argv[])
 {
+    // Store the program name from argv[0]
+    if (argc > 0 && argv[0]) {
+        program_name = argv[0];
+    } else 
+    {
+        fatal_error("Main: Could not determine program name from argv[0]\n");
+        return -1;
+    }
+
     gmloader_config.init_defaults();
 
     fs::path work_dir, config_file_path, save_dir, apk_path;
     work_dir = fs::canonical(fs::current_path()) / "";
 
-    if (argc > 2 && strcmp(argv[1], "-c") == 0) {
-        
-        config_file_path = work_dir / argv[2];
-
-        if(gmloader_config.parse_file(config_file_path.c_str()) < 0 ){
-            warning("Error while loading the config file\n");
+    // Check for -a (apk_path override) and -c (config file)
+    std::string override_apk_path;
+    for (int i = 1; i < argc; ++i) {
+        if (strcmp(argv[i], "-a") == 0 && i + 1 < argc) {
+            override_apk_path = argv[i + 1];
+            warning("Main: Using apk_path override from args: '%s'\n", override_apk_path.c_str());
+            i++; // Skip the value
+        }
+        else if (strcmp(argv[i], "-c") == 0 && i + 1 < argc) {
+            config_file_path = work_dir / argv[i + 1];
+            if (gmloader_config.parse_file(config_file_path.c_str()) < 0) {
+                warning("Error while loading the config file\n");
+            }
```

Comment:
Not critical as the config filename will be skip anyway on the next iteration but looks cleaner :D
```suggestion
            }
            i++; // Skip the value
```